### PR TITLE
Improve support for reverse proxy with nginx.

### PR DESCRIPTION
**Pull request by ChanthMiao**
I need to deploy geminabox behind nginx under a subdirectory (e.g. /ruby), which is not supported.
So, I try to made it by modifying geminabox as this commit.

### CODE REVIEW

**Comment by tnir** [comment]
@ChanthMiao Can you update changes as some files are conflicted?

**Comment by ChanthMiao** [comment]
> @ChanthMiao Can you update changes as some files are conflicted?

I may update this PR next week.

**Comment by ChanthMiao** [comment]
There are some checks failed. I need some time to fix it.

File: lib/geminabox/server.rb [hunk 1]
```diff
@@ -29,6 +29,10 @@ def allow_upload?
         Geminabox.allow_upload
       end
 
+      def base_path?
+        Geminabox.base_path.gsub(/\/$/,'')
```
[the remark this answers is not in the document]
Comment (ChanthMiao) [reply lib/geminabox/server.rb]: Already fixed.

**Comment by ChanthMiao** [comment]
@tnir please review this PR again.

File: lib/geminabox/server.rb [hunk 2]
```diff
@@ -29,6 +29,10 @@ def allow_upload?
         Geminabox.allow_upload
       end
 
+      def base_path?
+        Geminabox.base_path.sub(%r{/\/$/}, '')
```
Comment (ChanthMiao) [inline lib/geminabox/server.rb:33]: > lib/geminabox/server.rb:33:37: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
        Geminabox.base_path.sub(%r{/\/$/}, '')
                                    ^^

@tnir Any suggestions? I am not a ruby regex professional.

Comment (ChanthMiao) [reply lib/geminabox/server.rb]: It should be fixed now.
```diff
diff --git a/lib/geminabox/server.rb b/lib/geminabox/server.rb
index 7eecc2a..4cc8a5b 100644
--- a/lib/geminabox/server.rb
+++ b/lib/geminabox/server.rb
@@ -30,7 +30,7 @@ module Geminabox
       end

       def base_path?
-        Geminabox.base_path.sub(%r{/\/$/}, '')
+        Geminabox.base_path.sub(%r{/$}, '')
       end

       def fixup_bundler_rubygems!
```

**Comment by ChanthMiao** [comment]
@tnir, this time rubocop check should be ok!

**Review by tnir — CHANGES_REQUESTED** [review]
@ChanthMiao Would you mind adding tests for code you added? 🍀

**Comment by ChanthMiao** [comment]
> @ChanthMiao Would you mind adding tests for code you added? 🍀

I can try it, but have no idea how to do it, currently.
This PR changes the urls in web pages. which only works (and make sense) behind a reserve proxy. I just do not know how to tests it in testscases.

**Comment by ChanthMiao** [comment]
Maybe, we cant test it by running docker-compose in Github Actions. (e.g. <https://github.com/peter-evans/docker-compose-actions-workflow>)

@tnir any better suggestions?

**Pull request closed without merging**